* grobian@gentoo.org: uname -m on Solaris/Intel returns i86pc for both
                      x86 and x64

--- configure
+++ configure
@@ -1327,12 +1327,12 @@
 check_cflags -std=c99
 
 case "$arch" in
-    i386|i486|i586|i686|i86pc|BePC)
+    i386|i486|i586|i686|BePC)
         arch="x86"
         subarch="x86_32"
         enable fast_unaligned
     ;;
-    x86_64|amd64)
+    x86_64|amd64|i86pc)
         arch="x86"
         subarch="x86_32"
         enable cmov
